packages/teams-js/src/public/dialog.ts

```diff
@@ -111,183 +88,383 @@ export namespace dialog {
+      title: adaptiveCardDialogInfo.title,
+    };
+    return dialogInfo;
+  }
 
-    // We need to remove the original 'messageForChild'
-    // handler since the original does not allow for post messages.
-    // It is replaced by the user specified listener that is passed in.
-    removeHandler('messageForChild');
-    registerHandler('messageForChild', listener);
-    storedMessages.reverse();
-    while (storedMessages.length > 0) {
-      const message = storedMessages.pop();
-      listener(message);
-    }
+  /**
+   * @hidden
+   * Hide from docs
+   * --------
+   * Convert BotUrlDialogInfo to DialogInfo to send the information to host in {@linkcode url.bot.open} API.
+   *
+   * @internal
+   */
+  export function getDialogInfoFromBotUrlDialogInfo(botUrlDialogInfo: BotUrlDialogInfo): DialogInfo {
+    const dialogInfo: DialogInfo = {
+      url: botUrlDialogInfo.url,
+      height: botUrlDialogInfo.size ? botUrlDialogInfo.size.height : DialogDimension.Small,
+      width: botUrlDialogInfo.size ? botUrlDialogInfo.size.width : DialogDimension.Small,
+      title: botUrlDialogInfo.title,
+      fallbackUrl: botUrlDialogInfo.fallbackUrl,
+      completionBotId: botUrlDialogInfo.completionBotId,
+    };
+    return dialogInfo;
   }
 
   /**
-   * Checks if dialog module is supported by the host
+   * @hidden
+   * Hide from docs
+   * --------
+   * Convert AdaptiveCardDialogInfo to DialogInfo to send the information to host in {@linkcode adaptiveCard.open} API.
    *
-   * @returns boolean to represent whether dialog module is supported
+   * @internal
    */
-  export function isSupported(): boolean {
-    return runtime.supports.dialog ? true : false;
+  export function getDialogInfoFromBotAdaptiveCardDialogInfo(
+    botAdaptiveCardDialogInfo: BotAdaptiveCardDialogInfo,
+  ): DialogInfo {
+    const dialogInfo: DialogInfo = {
+      card: botAdaptiveCardDialogInfo.card,
+      height: botAdaptiveCardDialogInfo.size ? botAdaptiveCardDialogInfo.size.height : DialogDimension.Small,
+      width: botAdaptiveCardDialogInfo.size ? botAdaptiveCardDialogInfo.size.width : DialogDimension.Small,
+      title: botAdaptiveCardDialogInfo.title,
+      completionBotId: botAdaptiveCardDialogInfo.completionBotId,
+    };
+    return dialogInfo;
   }
 
   /**
-   * Namespace to update the dialog
+   * Subcapability for interacting with adaptive card dialogs
    */
-  export namespace update {
+  export namespace adaptiveCard {
     /**
-     * Update dimensions - height/width of a dialog.
+     * Allows app to open an adaptive card based dialog.
      *
-     * @param dimensions - An object containing width and height properties.
+     * @remarks
+     * This function cannot be called from inside of a dialog
+     *
+     * @param adaptiveCardDialogInfo - An object containing the parameters of the dialog module.
+     * @param submitHandler - Handler that triggers when a dialog calls the {@linkcode submit} function or when the user closes the dialog.
```

Comment:
what does "when a dialog calls the submit function" mean here? Because for AC based dialogs, there is no dev-provided JS running at all, is there?

Reply:
For adaptive-card based dialogs, this would be called when your adaptive-card fires the 'Action.Submit' action https://adaptivecards.io/explorer/Action.Submit.html

Will update the documentation here to be more clear

Reply:
(Will also move dialog.submit to be dialog.url.submit)